doc/formats/morphology.rst

```diff
@@ -0,0 +1,142 @@
+Arbor supports reading morphologies described using the
+`SWC <http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html>`_ file format.
+
+SWC files may contain comments, which are stored as metadata. And a blank line anywhere in the file is
+interpreted as end of data.
+
+The description of the morphology is encoded as a list of samples with an id,
+an `x,y,z` location in space, a radius, a tag and a parent id. Arbor parses these samples, performs some checks,
+then generates a morphology according to one of three possible interpretations.
+
+The SWC file format specifications are not very detailed, which has lead different simulators to interpret
+SWC files in different ways, especially when it comes to the soma. Arbor has its own an interpretation that
+is powerful and simple to understand at the same time. However, we have also developed functions that will
+interpret SWC files similarly to how the NEURON simulator would, and how the Allen Institute would.
+
+Despite the differences between the interpretations, there is a common set of checks that are always performed
+to validate an SWC file:
+
+* Check that there are no duplicate ids.
+* Check that the parent id of a sample is less than the id of the sample.
+* Check that the parent id of a sample refers to an existing sample.
+
+In addition, all interpretations agree that a *segment* is (in the common case) constructed between a sample and
+its parent and inherits the tag of the sample; and if more than 1 sample have the same parent, the parent sample
+is interpreted as a fork point in the morphology, and acts as the proximal point to a new branch for each of its
+"child" samples. There a couple of exceptions to these rules which are listed below.
+
+Arbor interpretation:
+"""""""""""""""""""""
+In addition to the previously listed checks, the arbor interpretation explicitly disallows SWC files where the soma is
+described by a single sample. It constructs the soma from 2 or more samples, forming 1 or more segments. A *segment* is
+always constructed between a sample and its parent. This means that there are no gaps in the resulting morphology.
+
+Arbor has no magic rules or transformations for the soma. It can be a single branch or multiple branches; segments
+of a different tag can connect to its distal end, proximal end or anywhere in the middle. For example, to create a
+morphology with a single segment soma; a single segment axon connected to one end of the soma; and a single segment
+dendrite connected to the other end of the soma, the following swc file can be used:
+
+
+.. literalinclude :: example.swc
+   :language: python
+   :linenos:
+
+Samples 1 and 2 will form the soma; samples 1 and 3 will form the axon, connected to the soma at the proximal end;
+samples 2 and 4 will form the dendrite, connected to the soma at the distal end. The morphology will look something
+like this:
+
+.. figure:: ../gen-images/swc_morph.svg
+   :width: 400
+   :align: center
+
```

Comment:
Maybe it's worth adding a note here saying something to the effect of this
- morphologies are abstract things, *not* 3d physical objects ie neither the given positions nor relative locations have any effect on the simulation
- we _render_ them like this for demonstration.